.github/workflows/vendor_release.yml

```diff
@@ -0,0 +1,24 @@
+name: Generate vendored content for the release
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  generate-and-publish-vendored-code:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Generate and publish cargo vendored tarball
+        run: |
+          tag=$(echo $GITHUB_REF | sed 's|refs/tags/||')
+          tarball_name="guest-components-${tag}-vendor.tar.gz"
+          [ -d .cargo ] || mkdir .cargo
+          cargo vendor >> .cargo/config.toml
```

Comment:
`mkdir -p` , but I assume we do want to abort if that folder exists, otherwise we'd archive unexpected stuff.

`cargo vendor --locked` probably makes sense here?

Reply:
https://github.com/fidencio/guest-components/releases/tag/v0.0.2-test-vendored-tarball